rfc/rfc-0020-resources-update-after-success.md

```diff
@@ -0,0 +1,91 @@
+# Draft RFC 20 Update Resources Only After Success/Failure
+
+## Summary
+
+Cartographer is currently unable to determine when an input to a resource has reached success, failed, or is still
+processing. Making Carto aware of status indications is not sufficient, as frequent updates to a resource could keep it
+in a constant state of 'processing', even as it succeeds and fails on multiple successive inputs. Carto should wait
+until an input has resulted in success or failure before updating the resource with new input. 
```

Comment:
> Carto should wait until an input has resulted in success or failure before updating the resource with new input

This assumes an arbitrary resource will always reach a "success" or "failure" state. If such a state is never reached, the supply chain is now deadlocked.

Reply:
1. I agree that for some objects the concept of success is meaningless. The [section on non-reconciling objects](https://github.com/vmware-tanzu/cartographer/blob/rfc-0020-template-success/rfc/rfc-0020-resources-update-after-success.md#handling-non-reconciling-objects) is meant to address those resource types. I'm interested in your reaction to the alternative approaches there.
2. I may have misinterpreted your comment. Perhaps you mean, some resource becomes deadlocked, that could deadlock the entire chain. To such an objection I would say, misconfiguration of templates or bad values fed in to a resource could indeed gum up the works. But I do not think that is a critique of the works.

Reply:
> Perhaps you mean, some resource becomes deadlocked, that could deadlock the entire chain. To such an objection I would say, misconfiguration of templates or bad values fed in to a resource could indeed gum up the works. But I do not think that is a critique of the works.

I'm not talking about a misconfiguration of Cartographer resources, but the behavior of the resource being choreographed.

It becomes Cartographer's problem when a supply chain is no longer able to make progress for a workload until the workload is deleted. If Cartographer is going to be usable in the real world, it will need to gracefully handle other resources that don't always behave perfectly.

Reply:
From a risk perspective, what is the difference between:
- a resource that is expected to report "success", but misbehaves and never does
- a resource that is expected to report an output, but misbehaves and never does

I think that we currently face the risk of the latter. And I think it is functionally equivalent to the former. So it doesn't seem like this RFC takes on more risk of deadlock.